src/app/OperationalDeviceProxy.cpp

```diff
@@ -105,11 +106,19 @@ CHIP_ERROR OperationalDeviceProxy::Connect(Callback::Callback<OnDeviceConnected>
         if (!isConnected)
         {
             err = LookupPeerAddress();
```

Comment:
Not defined, but I guess it could.  I guess the smart thing to do is to change state, then on error change it back.

I'll make that change.  I hate sync callbacks.

Reply:
Fixed.